### PR TITLE
Added forecastDatasetEncrypted plugin and spec

### PR DESCRIPTION
**Pull request by m-nouman-suleman**
## Task
https://scalock.atlassian.net/browse/SAAS-3258

### CODE REVIEW

File: plugins/aws/forecast/datasetExportEncrypted.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+module.exports = {
+    title: 'Forecast Dataset Export Encrypted',
+    category: 'Forecast',
+    domain: 'Content Delivery',
+    description: 'Ensure that AWS Forecast exports have encryption enabled before they are being saved on S3.',
+    more_info: 'In AWS Forecast, you can save forecast reports on S3 in CSV format. Make sure to encrypt these export before writing them to the bucket in order to follow your organizations\'s security and compliance requirements.',
+    recommended_action: 'Create Forecast exports with encryption enabled',
+    link: 'https://docs.aws.amazon.com/forecast/latest/dg/howitworks-forecast.html',
+    apis: ['ForecastService:listForecastExportJobs', 'KMS:listKeys', 'KMS:describeKey'],
+    settings: {},
```
Comment (AkhtarAmir) [inline plugins/aws/forecast/datasetExportEncrypted.js:13]: Add setting for encryption level

File: plugins/aws/forecast/datasetExportEncrypted.spec.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+var expect = require('chai').expect;
+var datasetExportEncrypted = require('./datasetExportEncrypted');
+
+const listForecastExportJobs = [
+    {
+        "ForecastExportJobArn": "arn:aws:forecast:us-east-1:101363889637:forecast-export-job/akhtar_fc/ewd",
```
Comment (AkhtarAmir) [inline plugins/aws/forecast/datasetExportEncrypted.spec.js:6]:
```suggestion
        "ForecastExportJobArn": "arn:aws:forecast:us-east-1:000011112222:forecast-export-job/akhtar_fc/ewd",
```

File: plugins/aws/forecast/datasetExportEncrypted.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                return rcb();
+            }
+
+            if (!listForecastExportJobs.data.length) {
+                helpers.addResult(results, 0, 'No forecast export found', region);
+                return rcb();
+            }
+
+            for (let forecastExportJob of listForecastExportJobs.data) {
+                let { S3Config } = forecastExportJob.Destination;
```
Comment (AkhtarAmir) [inline plugins/aws/forecast/datasetExportEncrypted.js:38]: add proper check

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/forecast/datasetExportEncrypted.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+            for (let forecastExportJob of listForecastExportJobs.data) {
+                if (!forecastExportJob && !forecastExportJob.Destination) {
+                    continue;
+                }
+
+                let { S3Config } = forecastExportJob.Destination;
+                let resource = forecastExportJob.ForecastExportJobArn;
+
+                if (S3Config && S3Config.KMSKeyArn) {
```
Comment (AkhtarAmir) [inline plugins/aws/forecast/datasetExportEncrypted.js:68]: Check if export is enabled and give passing result

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/forecast/datasetExportEncrypted.js (hunk 4 above)
Comment (m-nouman-suleman) [reply plugins/aws/forecast/datasetExportEncrypted.js:68]: listForecastExportJobs will only list active export jobs that have been enabled by the User so we do not have to check it

File: plugins/aws/forecast/datasetExportEncrypted.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+        async.each(regions.forecastservice, function(region, rcb){
+            var listForecastExportJobs = helpers.addSource(cache, source,
+                ['forecastservice', 'listForecastExportJobs', region]);
+
+            if (!listForecastExportJobs) return rcb();
+
+            if (listForecastExportJobs.err || !listForecastExportJobs.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query forecast exports: ' + helpers.addError(listForecastExportJobs), region);
```
Comment (AkhtarAmir) [inline plugins/aws/forecast/datasetExportEncrypted.js:42]:
```suggestion
                    'Unable to query Forecast exports: ' + helpers.addError(listForecastExportJobs), region);
```

File: plugins/aws/forecast/datasetExportEncrypted.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            if (!listForecastExportJobs) return rcb();
+
+            if (listForecastExportJobs.err || !listForecastExportJobs.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query Forecast exports: ' + helpers.addError(listForecastExportJobs), region);
+                return rcb();
+            }
+
+            if (!listForecastExportJobs.data.length) {
+                helpers.addResult(results, 0, 'No forecast exports found', region);
```
Comment (AkhtarAmir) [inline plugins/aws/forecast/datasetExportEncrypted.js:47]:
```suggestion
                helpers.addResult(results, 0, 'No Forecast exports found', region);
```

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/aws/forecast/datasetExportEncrypted.spec.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+            datasetExportEncrypted.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if unable to find Wisdom domain encryption key', function (done) {
+            const cache = createCache([listForecastExportJobs[0]], null, null);
```
Comment (AkhtarAmir) [inline plugins/aws/forecast/datasetExportEncrypted.spec.js:154]:
```suggestion
            const cache = createCache([listForecastExportJobs[1]], listKeys, null);
```

**Pull request merged**